controllers/limitador_cluster_envoyfilter_controller.go

```diff
@@ -0,0 +1,170 @@
+/*
+Copyright 2021 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	limitadorv1alpha1 "github.com/kuadrant/limitador-operator/api/v1alpha1"
+	istioapinetworkingv1alpha3 "istio.io/api/networking/v1alpha3"
+	istioclientnetworkingv1alpha3 "istio.io/client-go/pkg/apis/networking/v1alpha3"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	meta "k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	gatewayapiv1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
+
+	"github.com/kuadrant/kuadrant-operator/pkg/common"
+	kuadrantistioutils "github.com/kuadrant/kuadrant-operator/pkg/istio"
+	"github.com/kuadrant/kuadrant-operator/pkg/reconcilers"
+)
+
+// LimitadorClusterEnvoyFilterReconciler reconciles a EnvoyFilter object with limitador's cluster
+type LimitadorClusterEnvoyFilterReconciler struct {
+	*reconcilers.BaseReconciler
+}
+
+//+kubebuilder:rbac:groups=networking.istio.io,resources=envoyfilters,verbs=get;list;watch;create;update;patch;delete
+
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.10.0/pkg/reconcile
+func (r *LimitadorClusterEnvoyFilterReconciler) Reconcile(eventCtx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logger := r.Logger().WithValues("Gateway", req.NamespacedName)
+	logger.Info("Reconciling EnvoyFilter")
+	ctx := logr.NewContext(eventCtx, logger)
+
+	gw := &gatewayapiv1beta1.Gateway{}
+	if err := r.Client().Get(ctx, req.NamespacedName, gw); err != nil {
+		if apierrors.IsNotFound(err) {
+			logger.Info("no gateway found")
+			return ctrl.Result{}, nil
+		}
+		logger.Error(err, "failed to get gateway")
+		return ctrl.Result{}, err
+	}
+
+	if logger.V(1).Enabled() {
+		jsonData, err := json.MarshalIndent(gw, "", "  ")
```

Comment:
Why `MarshalIndent`? I wonder whether it's suitable for when using structured logging, aka `production` log mode.

Reply:
AFAIK, from the `go-logr` interface I can only check logger level, nothing about the format (defined by development or production mode). I see that this `MarshalIndent` call is not entirely correct as logger could be configured with structured format and debug level. By default (no `LOG_*` env vars), the logger is configured with `info` level and `production` level. When running with `make run`. it is `debug` and `development`. Thus, no issue with this so far. 

Any better approach you are considering? Not for this PR, but considering that (almost) all controllers have this code in the beginning of the `Reconcile` method, it would be nice to have all of them fixed.

Reply:
I didn't mean checking here what the log mode setting is. I just assumed that log level and log mode were two separate things, like in [Authorino](https://docs.kuadrant.io/authorino/docs/user-guides/observability/#log-levels-and-log-modes). But it looks like log mode and log level are somehow coupled together.

IMO, it does not have to be like that.

In Authorino, all 4 combinations are allowed, i.e.:
- `level: info, mode: production`
- `level: debug, mode: production`
- `level: info, mode: development`
- `level: debug, mode: production`

(Maybe the naming is a bit unfortunate, because `production` and `development` mean actually "structured" and "human-readable", respectively.)

Because it felt weird printing indented JSON when log mode is set to `production`, I'd probably have used `json.Marshal` instead. Not a big deal though.

Reply:
This project's log system is exactly the same, with those 4 combinations allowed. 

> Because it felt weird printing indented JSON when log mode is set to production, I'd probably have used json.Marshal instead. Not a big deal though.

Check my previous comment out again. You are right. However, if I change to `Mashal`, when I run in debug + development mode when I run with `make run`, it is not human readable. I need to indent it :)

So looking for alternatives that makes a good dev experience as well.